### PR TITLE
AG-11190 PoC Keyboard Navigation

### PR DESCRIPTION
**Pull request by olegat**
* https://ag-grid.atlassian.net/browse/AG-11107
* https://ag-grid.atlassian.net/browse/AG-11190

### CODE REVIEW

File: packages/ag-charts-community/src/chart/axis/axis.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ import { ChartAxisDirection } from '../chartAxisDirection';
 import { CartesianCrossLine } from '../crossline/cartesianCrossLine';
 import type { CrossLine } from '../crossline/crossLine';
 import type { AnimationManager } from '../interaction/animationManager';
-import type { InteractionEvent } from '../interaction/interactionManager';
+import type { PointerInteractionEvent as InteractionEvent } from '../interaction/interactionManager';
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/axis/axis.ts:47]: Nit: Might be better to import as the actual name of the interface, to save confusion later?

File: packages/ag-charts-community/src/chart/caption.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ import {
     Validate,
 } from '../util/validation';
 import type { CaptionLike } from './captionLike';
-import type { InteractionEvent } from './interaction/interactionManager';
+import type { PointerInteractionEvent as InteractionEvent } from './interaction/interactionManager';
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/caption.ts:20]: Ditto

File: packages/ag-charts-community/src/chart/chart.ts [hunk 3]
```diff
@@ -54,7 +54,7 @@ import { CursorManager } from './interaction/cursorManager';
 import { GestureDetector } from './interaction/gestureDetector';
 import type { HighlightChangeEvent } from './interaction/highlightManager';
 import { HighlightManager } from './interaction/highlightManager';
-import type { InteractionEvent, PointerOffsets } from './interaction/interactionManager';
+import type { PointerInteractionEvent as InteractionEvent, PointerOffsets } from './interaction/interactionManager';
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/chart.ts:57]: Ditto

File: packages/ag-charts-community/src/chart/interaction/keyNavManager.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+export type KeyNavEvent<T extends KeyNavEventType = KeyNavEventType> = {
+    type: T;
+    delta: number;
+    interactionEvent: InteractionEvent;
+};
+
+export class KeyNavManager extends BaseManager<KeyNavEventType, KeyNavEvent> {
+    private isFocused: boolean = false;
+    private readonly destroyFns: (() => void)[] = [];
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/interaction/keyNavManager.ts]: Is it worth pushing this up to `BaseManager` as a responsibility?

File: packages/ag-charts-community/src/chart/legend.ts [hunk 5]
```diff
@@ -36,7 +36,7 @@ import {
 import { ChartUpdateType } from './chartUpdateType';
 import type { Page } from './gridLayout';
 import { gridLayout } from './gridLayout';
-import { InteractionEvent, InteractionState } from './interaction/interactionManager';
+import { PointerInteractionEvent as InteractionEvent, InteractionState } from './interaction/interactionManager';
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/legend.ts:39]: Ditto

File: packages/ag-charts-community/src/chart/navigator/navigator.ts [hunk 6]
```diff
@@ -7,7 +7,7 @@ import { Logger } from '../../util/logger';
 import { clamp } from '../../util/number';
 import { ActionOnSet, ObserveChanges } from '../../util/proxy';
 import { AND, BOOLEAN, GREATER_THAN, LESS_THAN, OBJECT, POSITIVE_NUMBER, RATIO, Validate } from '../../util/validation';
-import { InteractionEvent, InteractionState } from '../interaction/interactionManager';
+import { PointerInteractionEvent as InteractionEvent, InteractionState } from '../interaction/interactionManager';
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/navigator/navigator.ts:10]: Ditto

File: packages/ag-charts-community/src/chart/pagination/pagination.ts [hunk 7]
```diff
@@ -17,7 +17,7 @@ import {
 } from '../../util/validation';
 import { ChartUpdateType } from '../chartUpdateType';
 import type { CursorManager } from '../interaction/cursorManager';
-import type { InteractionEvent } from '../interaction/interactionManager';
+import type { PointerInteractionEvent as InteractionEvent } from '../interaction/interactionManager';
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/pagination/pagination.ts:20]: Ditto

File: packages/ag-charts-community/src/chart/tooltip/tooltip.ts [hunk 8]
```diff
@@ -15,7 +15,7 @@ import {
     UNION,
     Validate,
 } from '../../util/validation';
-import type { InteractionEvent, PointerOffsets } from '../interaction/interactionManager';
+import type { PointerInteractionEvent as InteractionEvent, PointerOffsets } from '../interaction/interactionManager';
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/tooltip/tooltip.ts:18]: Ditto

File: packages/ag-charts-community/src/chart/axis/axis.ts (hunk 1 above)
Comment (olegat) [reply packages/ag-charts-community/src/chart/axis/axis.ts:47]: Agreed. I've got a follow up PR for this nit https://github.com/ag-grid/ag-charts/pull/1361/files. This was just a temporary workaround.

File: packages/ag-charts-community/src/chart/caption.ts (hunk 2 above)
Comment (olegat) [reply packages/ag-charts-community/src/chart/caption.ts:20]: https://github.com/ag-grid/ag-charts/pull/1361/files

File: packages/ag-charts-community/src/chart/chart.ts (hunk 3 above)
Comment (olegat) [reply packages/ag-charts-community/src/chart/chart.ts:57]: https://github.com/ag-grid/ag-charts/pull/1361/files

File: packages/ag-charts-community/src/chart/legend.ts (hunk 5 above)
Comment (olegat) [reply packages/ag-charts-community/src/chart/legend.ts:39]: https://github.com/ag-grid/ag-charts/pull/1361/files

File: packages/ag-charts-community/src/chart/navigator/navigator.ts (hunk 6 above)
Comment (olegat) [reply packages/ag-charts-community/src/chart/navigator/navigator.ts:10]: https://github.com/ag-grid/ag-charts/pull/1361/files

File: packages/ag-charts-community/src/chart/pagination/pagination.ts (hunk 7 above)
Comment (olegat) [reply packages/ag-charts-community/src/chart/pagination/pagination.ts:20]: https://github.com/ag-grid/ag-charts/pull/1361/files

File: packages/ag-charts-community/src/chart/tooltip/tooltip.ts (hunk 8 above)
Comment (olegat) [reply packages/ag-charts-community/src/chart/tooltip/tooltip.ts:18]: https://github.com/ag-grid/ag-charts/pull/1361/files

File: packages/ag-charts-community/src/chart/interaction/keyNavManager.ts (hunk 4 above)
Comment (olegat) [reply packages/ag-charts-community/src/chart/interaction/keyNavManager.ts]: Done. https://github.com/ag-grid/ag-charts/commit/e6e7942a8043e38f0decd154b34dd2e177092212

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**